common/determined_common/experimental/checkpoint/_tf.py

```diff
@@ -1,45 +1,52 @@
 import pathlib
-from typing import List, Optional
+from typing import Any, Dict, List, Optional, cast
 
 import tensorflow as tf
 from tensorflow.python.training.tracking.tracking import AutoTrackable
 
+from determined.experimental._native import _local_trial_from_context
+from determined.keras import TFKerasTrial
 
-def load_model(ckpt_dir: pathlib.Path, tags: Optional[List[str]] = None) -> AutoTrackable:
-    saved_model_paths = list(ckpt_dir.glob("**/saved_model.pb"))
-    h5_paths = list(ckpt_dir.glob("**/*.h5"))
 
-    if not h5_paths and not saved_model_paths:
-        raise AssertionError(
-            "No checkpoint saved_model.pb or h5 files found at {}".format(ckpt_dir)
+def load_model(
+    ckpt_dir: pathlib.Path, metadata: Dict[str, Any], tags: Optional[List[str]] = None
+) -> AutoTrackable:
+    save_format = metadata.get("format", "")
+
+    if save_format == "saved_model" or "format" not in metadata:
+        return load_saved_model(ckpt_dir, tags=tags)
+
+    elif save_format == "h5":
+        trial = _local_trial_from_context(
+            ckpt_dir.joinpath("code"),
+            config=metadata["experiment_config"],
+            hparams=metadata["hparams"],
         )
 
+        trial = cast(TFKerasTrial, trial)
+        model = trial.build_model()
+        model.load_weights(str(ckpt_dir.joinpath("determined-keras-model.h5")))
+        return model
+
```

Comment:
non-blocking: for completeness probably good to add a check here for unknown `save_format`